src/poetry/installation/chef.py

```diff
@@ -41,15 +41,10 @@ def should_prepare(self, archive: Path) -> bool:
     def is_wheel(self, archive: Path) -> bool:
         return archive.suffix == ".whl"
 
-    def get_cached_archive_for_link(self, link: Link) -> Link:
-        # If the archive is already a wheel, there is no need to cache it.
-        if link.is_wheel:
-            return link
```

Comment:
Considering the comment, I can only imagine that might be a shortcut for local wheels? Maybe, you can examine if the change has some side effects on wheels as path dependencies. 🙏

Reply:
The only place this is called is in the executor `_download_link()` code path, which means that it is never relevant for path dependencies.

I'm quite unsure about what this is supposed to be doing - `Chef` isn't a name that helps much.  At first I'd thought it was going to be some general purpose artifact cache, but its extremely limited scope is puzzling to me.

Reply:
I need to look at this when I have more time. But I suspect the original intent was to cache the environment specific when built. Useful when the link is an sdist.

Reply:
https://github.com/python-poetry/poetry/pull/5871#issuecomment-1168805110